contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,285 @@
+        external
+        returns(uint256)
+    {
+        I_PerpetualV1 perpetual = I_PerpetualV1(_PERPETUAL_V1_);
+
+        // Get the balances of the sender.
+        perpetual.deposit(msg.sender, 0);
+        P1Types.Balance memory initialBalance = perpetual.getAccountBalance(msg.sender);
+
+        // Get the maximum liquidatable amount.
+        SignedMath.Int memory maxPositionDelta = _getMaxPositionDelta(
+            initialBalance,
+            isBuy,
+            maxPosition
+        );
+
+        // Do the liquidation.
+        _doLiquidation(
+            perpetual,
+            liquidatee,
+            maxPositionDelta
+        );
+
+        // Get the balances of the sender.
+        P1Types.Balance memory currentBalance = perpetual.getAccountBalance(msg.sender);
+
+        // Get the liquidated amount and fee amount.
+        (uint256 liqAmount, uint256 feeAmount) = _getLiquidatedAndFeeAmount(
+            perpetual,
+            isBuy,
+            initialBalance,
+            currentBalance
+        );
+
+        // Transfer fee from sender to insurance fund.
+        perpetual.withdraw(msg.sender, address(this), feeAmount);
+        perpetual.deposit(_INSURANCE_FUND_, feeAmount);
+
+        // Log the result.
+        emit LogLiquidatorProxyUsed(
+            msg.sender,
+            liquidatee,
+            isBuy,
+            liqAmount,
+            feeAmount
+        );
+
+        return liqAmount;
+    }
+
+    // ============ Admin Functions ============
+
+    /**
+     * @dev Allows the owner to set the insurance fund address. Emits the LogInsuranceFundSet event.
+     *
+     * @param  insuranceFund  The address to set as the insurance fund.
+     */
+    function setInsuranceFund(
+        address insuranceFund
+    )
+        external
+        onlyOwner
+    {
+        _INSURANCE_FUND_ = insuranceFund;
+        emit LogInsuranceFundSet(insuranceFund);
+    }
+
+    // ============ Helper Functions ============
+
+    /**
+     * @dev Calculate (and verify) the maximum amount to liquidate based on the maxPosition input.
+     */
+    function _getMaxPositionDelta(
+        P1Types.Balance memory initialBalance,
+        bool isBuy,
+        SignedMath.Int memory maxPosition
+    )
+        private
+        pure
+        returns (SignedMath.Int memory)
+    {
+        SignedMath.Int memory result = maxPosition.signedSub(initialBalance.getPosition());
+
+        require(
+            result.isPositive == isBuy && result.value > 0,
+            "Cannot liquidate past maxPosition"
+        );
+
+        return result;
+    }
+
+    /**
+     * @dev Perform the liquidation by constructing the correct arguments and sending it to the
+     * perpetual.
+     */
+    function _doLiquidation(
+        I_PerpetualV1 perpetual,
+        address liquidatee,
+        SignedMath.Int memory maxPositionDelta
+    )
+        private
+    {
+        // Create accounts.
+        bool takerFirst = address(msg.sender) < liquidatee;
```

Comment:
what is this? Why are you ordering which trade goes first based on comparing the addresses? Wouldn't it be easier to always have a consistent ordering (e.g. msg.sender is always accounts[0])?

Reply:
The accounts must be sorted

Reply:
To do less calculations ensuring that the accounts are unique